server/src/main/java/org/opensearch/search/SearchSortValues.java

```diff
@@ -67,6 +67,11 @@ public class SearchSortValues implements ToXContentFragment, Writeable {
         this.rawSortValues = EMPTY_ARRAY;
     }
 
+    public SearchSortValues(Object[] sortValues, Object[] rawSortValues) {
```

Comment:
This is scary (it seems like we cannot make this constructor package private), but a least

```
if (rawSortValues.length != sortValues.length) {
            throw new IllegalArgumentException("rawSortValues and sortValues must hold the same number of items");
        }
        }
```

Reply:
Also, please handle case when `rawSortValues` could be `null`